src/views/catalog/CreateFromInstanceTypes/components/BootableVolumeList/components/BootableVolumeRow/BootableVolumeRow.tsx

```diff
@@ -2,45 +2,67 @@ import React, { FC } from 'react';
 
 import { getTemplateOSIcon as getOSIcon } from '@catalog/templatescatalog/utils/os-icons';
 import { V1beta1DataSource } from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
-import { V1alpha2VirtualMachineClusterPreference } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import {
+  V1alpha1PersistentVolumeClaim,
+  V1alpha2VirtualMachineClusterPreference,
+} from '@kubevirt-ui/kubevirt-api/kubevirt';
 import { ANNOTATIONS } from '@kubevirt-utils/resources/template';
+import { NO_DATA_DASH } from '@kubevirt-utils/resources/vm/utils/constants';
 import { Text, TextVariants } from '@patternfly/react-core';
-import { Td, Tr } from '@patternfly/react-table';
+import { Tr } from '@patternfly/react-table';
+
+import { humanizeBinaryBytes } from '../../../../../../../utils/utils/humanize';
+
+import TableData from './TableData';
 
 type BootableVolumeRowProps = {
   bootableVolume: V1beta1DataSource;
+  activeColumnIDs: string[];
   rowData: {
     bootableVolumeSelectedState: [
       V1beta1DataSource,
       React.Dispatch<React.SetStateAction<V1beta1DataSource>>,
     ];
     preference: V1alpha2VirtualMachineClusterPreference;
+    pvcSource: V1alpha1PersistentVolumeClaim;
   };
 };
 
 const BootableVolumeRow: FC<BootableVolumeRowProps> = ({
   bootableVolume,
+  activeColumnIDs,
   rowData: {
     bootableVolumeSelectedState: [bootableVolumeSelected, setBootSourceSelected],
     preference,
+    pvcSource,
   },
 }) => {
   const bootVolumeName = bootableVolume?.metadata?.name;
+  const pvcDiskSize = pvcSource?.spec?.resources?.requests?.storage;
+  const sizeData = pvcDiskSize && humanizeBinaryBytes(pvcDiskSize);
   return (
     <Tr
       isHoverable
       isSelectable
       isRowSelected={bootableVolumeSelected?.metadata?.name === bootVolumeName}
       onClick={() => setBootSourceSelected(bootableVolume)}
     >
-      <Td id="volume-name">
+      <TableData activeColumnIDs={activeColumnIDs} id="name">
         <img src={getOSIcon(preference)} alt="os-icon" className="vm-catalog-row-icon" />
         <Text component={TextVariants.small}>{bootVolumeName}</Text>
-      </Td>
-      <Td id="operating-system">{preference?.metadata?.annotations?.[ANNOTATIONS.displayName]}</Td>
-      <Td id={ANNOTATIONS.description}>
-        {bootableVolume?.metadata?.annotations?.[ANNOTATIONS.description]}
-      </Td>
+      </TableData>
+      <TableData activeColumnIDs={activeColumnIDs} id="operating-system">
+        {preference?.metadata?.annotations?.[ANNOTATIONS.displayName] || NO_DATA_DASH}
```

Comment:
Hmm, I was thinking, maybe you could use getPreferenceReadableOS I've created some time ago in here, and to update that function to get the preference object as an optional parameter, then to return what you need..And maybe to move that function to more common utils, to be used by both Bootable volumes list and also by instanceTypes tab in Catalog. Not sure if it worth. WDYT?

Reply:
I don't think that function will fit this case here as I just need to select a field, and I also need to drill down the preferences from the tabel to each row

Reply:
No problem, keep it as it is :)